E2E setup – don't install deprecated WooCommerce Blocks plugin

Jinksi opened the pull request:
Fixes #8291 

#### Changes proposed in this Pull Request

The [WooCommerce Blocks plugin has now been archived](https://developer.woo.com/2024/02/22/woocommerce-blocks-is-being-archived-on-github-and-wordpress-org/) and all functionality rolled into WooCommerce Core, causing the e2e setup script to fail.

Therefore, this PR removes this plugin installation from the E2E env setup script.

> [!CAUTION]
> This PR will merge to `trunk`, which is required to ensure this fix applies to `woocommerce-payments-server` GH checks.

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Ensure E2E GH checks pass the `Setup E2E environment` step. (Merchant E2E tests were failing prior to this for an unrelated reason – [see previous run](https://github.com/Automattic/woocommerce-payments/actions/runs/8046038729/job/21972483308?pr=8216))

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile **N/A**

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

### Post-merge
- [ ] Cherry-pick the merge commit to `develop`

Jinksi committed:
Don't attempt to install woo-gutenberg-products-block, which is now part of WC Core

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8290` or branch name `fix/failing-e2e-setup-woo-gutenberg-products-block` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/failing-e2e-setup-woo-gutenberg-products-block) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: a635fa95da67c548e508f434672f0a7ef60911b5
- Build time: 2024-02-29 03:25:14 UTC

_Note: the build is updated when a new commit is pushed to this PR._

brucealdridge commented on tests/e2e/env/setup.sh:
```diff
@@ -324,12 +324,6 @@ else
 	echo "Skipping install of Action Scheduler"
 fi
 
-if [[ ! ${SKIP_WC_BLOCKS_TESTS} ]]; then
```
If we're removing this, should we remove references to it as it no longer does anything.

Jinksi replied:
For this PR, I think it is best to leave in this env var because it is used to filter out some test suites, e.g:

https://github.com/Automattic/woocommerce-payments/blob/adae84b8d870865befd409f1466edbeef7993740/tests/e2e/utils/flows.js#L60

https://github.com/Automattic/woocommerce-payments/blob/adae84b8d870865befd409f1466edbeef7993740/tests/e2e/specs/blocks/shopper/shopper-wc-blocks-saved-card-checkout-and-usage.spec.js#L27

brucealdridge reviewed (approved):
Recommend removing other references to `SKIP_WC_BLOCKS_TESTS` and adding a changelog.

Jinksi committed:
Update comment to clarify `SKIP_WC_BLOCKS_TESTS` is not related to installation

Jinksi committed:
Add placeholder changelog entry

achyuthajoy reviewed (changes requested):
@Jinksi Changes look good to me. But the PR should be merged to trunk not develop. If merged to develop, server tests won't pick up the changes until next release.

Jinksi commented:
Bugger, changed target branch to `trunk` via GH UI and now have carried over unrelated `develop` changes into this PR.

Jinksi committed:
Merge branch 'trunk' into fix/failing-e2e-setup-woo-gutenberg-products-block

Jinksi committed:
Remove changelog file since merging to `trunk`

Jinksi commented:
I've rebased this PR and removed the erraneous commits

> Bugger, changed target branch to `trunk` via GH UI and now have carried over unrelated `develop` changes into this PR.

I've rebased and removed the erroneous commits.

I've also removed the changelog file which would be unsuitable for a merge to `trunk`, since this is not a user-facing change or patch release.

achyuthajoy reviewed (approved):
LGTM 🚢 

Thanks for working on this @Jinksi

Jinksi commented:
Confirmed that this fixed the issue for server PR GH checks - [example](https://github.com/Automattic/woocommerce-payments-server/actions/runs/8083149038/job/22109391985?pr=4819).
